### DISCUSSION TITLE
Recognize a nightly battery charge without GSM

### DISCUSSION THREAD

**Discussion by camuxx86** [flobz/psa_car_controller#271] [Ideas]
Hi Flobz, 
some time ago I used TRONITY to record the car's position, charge etc...
One useful function was also the possibility of recognize the nightly charge even without the GSM signal...

I explain better.

I back to home, the battery is, for example 35%, and it's my last value stored.
I charge the battery for all the night in my underground garage without GSM signal, 
The next day, I stop the charge, and then, I go out, the GSM signal is ok and the new battery value is now 80% .

Can you, recognize this increment of battery percentage, as a charge, and store it in the db?
You could estimate, as "start at " and "start level" as last battery timestamp and value, and "stop at" and "end level" as new battery timestamp and value.
The same for the field "consumption"...

Is it in your opinion a good idea?

**Comment by Magneticdud** [comment]
It already does like this. For some reason my car does not communicate with the servers anymore during AC charges, so all I see is a "jump" from 35% to 80%, in the log present as a charge

**Reply by CalicodesIles** [reply]
Yes this behaviour is detailed here too.

https://github.com/flobz/psa_car_controller/issues/276

**Reply by Magneticdud** [reply]
i think psa wants to save money in data fees from their iot gsm sim cards and they sent some command to disable telemetry during ac charging? Even from the official app i can't see the status anymore, which is extremely inconvenient

**Comment by CalicodesIles** [comment]
Are you sure ? In my case a simple refresh command during an AC charge and the value is refreshed in MyPeugeot and in PSA Car Controller.

**Reply by Magneticdud** [reply]
Yes, the only way to wake up the car and see the soc during ac charging is to trigger the hvac from the opel app

**Comment by camuxx86** [comment]
No in my case doesn't work,
I don't have GSM coverage in my garage, and all the "domestic recharges" are not shown...
I cannot trigger the refresh in my case.
Instead, I can see it in tronity app.
